lib/ocpp/v201/charge_point.cpp

```diff
@@ -80,7 +80,7 @@ ChargePoint::ChargePoint(const std::map<int32_t, int32_t>& evse_connector_struct
         EVLOG_AND_THROW(std::invalid_argument("All non-optional callbacks must be supplied"));
     }
 
-    this->device_model = std::make_unique<DeviceModel>(std::move(device_model_storage));
+    this->device_model = std::make_shared<DeviceModel>(std::move(device_model_storage));
```

Comment:
This was by my suggestion since that would be preferred over passing a reference to a `unique_ptr` into the `SmartChargingHandler`. Although I have been in some discussions, and would have to agree with them, that for most of these we don't need to pass the ptr at all and could just pas a reference to the `DeviceModel` in this case.

Reply:
Right, that is what I thought - why not just pass a reference? Again, I would suggest filing a cleanup issue, and bringing it up for discussion for future cleanup. Even if we don't make any changes, at least we should do so intentionally 😄